bp/src/Bluepill.m

```diff
@@ -600,7 +603,8 @@ - (void)finishWithContext:(BPExecutionContext *)context {
 
     if (![self hasRemainingTestsInContext:context] && (context.attemptNumber <= [context.config.errorRetriesCount integerValue])) {
         [BPUtils printInfo:INFO withString:@"No more tests to run."];
-        self.finalExitStatus = context.exitStatus;
+        [BPUtils printInfo:INFO withString:@"%s:%d finalExitStatus = %@", __FILE__, __LINE__, [BPExitStatusHelper stringFromExitStatus:self.finalExitStatus]];
+        self.finalExitStatus = context.finalExitStatus | context.exitStatus;
```

Comment:
Maybe add a comment that #338 was reverted while we investigate other problems.

Reply:
Done.

Reply:
Done.